yaksh/models.py

```diff
@@ -2632,4 +2649,38 @@ class TestCaseOrder(models.Model):
     # Order of the test case for a question.
     order = models.TextField()
 
+
 ##############################################################################
+class ForumBase(models.Model):
+    uid = models.UUIDField(unique=True, default=uuid.uuid4, editable=False)
+    creator = models.ForeignKey(User, on_delete=models.CASCADE)
+    description = models.TextField()
+    created_at = models.DateTimeField(auto_now_add=True)
+    modified_at = models.DateTimeField(auto_now=True)
+    image = models.ImageField(upload_to=get_image_dir, blank=True,
+                              null=True, validators=[validate_image])
+    active = models.BooleanField(default=True)
+
+
+class Post(ForumBase):
+    title = models.CharField(max_length=200)
+    course = models.ForeignKey(Course,
+                               on_delete=models.CASCADE, related_name='post')
+
+    def __str__(self):
+        return self.title
+
+    def get_last_comment(self):
+        return self.comment.last()
+
+    def get_comments_count(self):
+        return self.comment.filter(active=True).count()
+
+
+class Comment(ForumBase):
+    post_field = models.ForeignKey(Post, on_delete=models.CASCADE,
```

Comment:
can be simply ```post```

Reply:
It conflicts with the model name. Hence, post_field.